### PR TITLE
CallSites for MemoryExtensions.Contains (CoreCLR)

### PR DESCRIPTION
**Pull request by grant-d**
- See https://github.com/dotnet/corefx/issues/27526 : `Contains` is optimized over `IndexOf`
- Found additional (~12) callsites that can take advantage of `Span<T>.Contains`
- Extension method does not need to be `unsafe`

### CODE REVIEW

**Comment by grant-d** [comment]
@ahsonkhan, not sure what you are proposing with that one - it is an instance method on `string`, and delegates to `string.IndexOf` not `Span<T>.IndexOf`. Unless you are proposing I use `string.AsSpan<char>.Contains`?

**Comment by ahsonkhan** [comment]
> Unless you are proposing I use `string.AsSpan<char>.Contains`?

No, that isn't what I was suggesting.

> not sure what you are proposing with that one - it is an instance method on `string`, and delegates to `string.IndexOf` not `Span<T>.IndexOf`

Right now `public bool Contains(char value)` is calling the following, which ends up calling the SpanHelpers static method:
https://github.com/dotnet/coreclr/blob/91820bdd42711266faae1c15d2176b9f47e48efe/src/System.Private.CoreLib/shared/System/String.Searching.cs#L38

Is it possible for it to instead call the following, directly?
https://github.com/dotnet/coreclr/blob/f9c2e89f31f3816c921abb7c84e1951ce37d3127/src/System.Private.CoreLib/shared/System/SpanHelpers.Char.cs#L80

So, something like:
```C#
        public bool Contains(char value) => SpanHelpers.Contains(ref _firstChar, value, Length);
```

**Comment by grant-d** [comment]
Sorry, must be a Friday, that makes sense. 
PR updated.

**Comment by ahsonkhan** [comment]
@dotnet-bot test Ubuntu x64 Checked CoreFX Tests

**Comment by jkotas** [comment]
@dotnet-bot test Ubuntu x64 Checked CoreFX Tests

**Comment by grant-d** [comment]
@ahsonkhan would you mind taking another look at this, I have added a couple more call sites since your previous approval so it would be great if you could confirm it's kosher (& hopefully merge it)?

File: src/System.Private.CoreLib/src/System/Environment.cs [hunk 1]
```diff
@@ -420,7 +420,7 @@ private static void ValidateVariableAndValue(string variable, ref string value)
             {
                 throw new ArgumentException(SR.Argument_StringFirstCharIsZero, nameof(variable));
             }
-            if (variable.IndexOf('=') != -1)
+            if (variable.Contains('='))
```
[the remark this answers is not in the document]
Comment (grant-d) [reply src/System.Private.CoreLib/src/System/Environment.cs]: Fixed in separate PR

**Pull request merged**